### PR TITLE
Colormap Renderer

### PR DESCRIPTION
**Pull request by vquach2404**

### CODE REVIEW

File: arcgis-ios-sdk-samples/Layers/Colormap renderer/ColormapRendererViewController.swift [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ArcGIS
+
+class ColormapRendererViewController: UIViewController {
+    @IBOutlet private weak var mapView: AGSMapView! {
```
Comment (philium) [inline arcgis-ios-sdk-samples/Layers/Colormap renderer/ColormapRendererViewController.swift]: Some samples use access control to make properties and methods `private`, but it makes no practical difference. The default of `internal` is just fine.

File: arcgis-ios-sdk-samples/Layers/Colormap renderer/README.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# Colormap Renderer
+
+Apply a colormap renderer to a raster.
+
+![Colormap renderer](colormap-renderer.png)
+
+## Use Case
+
+A colormap renderer transforms pixel values in a raster to display raster data based on specific colors, aiding in visual analysis of the data. For example, a forestry commission may want to quickly visualize areas above and below the tree-line line occurring at a know elevation on a raster containing elevation values. They could overlay a transparent colormap set to color those areas below the tree-line elevation green, and those above white.
```
[the remark this answers is not in the document]
Comment (vquach2404) [reply arcgis-ios-sdk-samples/Layers/Colormap renderer/README.md]: This is how it was described in the sample design. Should I remove one "line"?

[the remark this answers is not in the document]
Comment (philium) [reply arcgis-ios-sdk-samples/Layers/Colormap renderer/README.md]: If it is a typo (it looks like one to me), create a PR to get the readme template updated. I would go ahead and update this PR as well. If you're not sure, ask in the issue or in Slack.

File: arcgis-ios-sdk-samples/Layers/Colormap renderer/README.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+* AGSRaster
+* AGSRasterLayer
+
+## About the data
+
+The raster used in this sample shows an area in the south of the Shasta-Trinity National Forest, California.
+
+## Offline data
+
+This sample uses the [ShastaBW](https://www.arcgis.com/home/item.html?id=cc68728b5904403ba637e1f1cd2995ae) raster file. It is downloaded from ArcGIS Online automatically.
```
Comment (philium) [inline arcgis-ios-sdk-samples/Layers/Colormap renderer/README.md]: > It is downloaded from ArcGIS Online automatically.

That is not correct; this data is included in the repository. At the very least I would remove this sentence. Other samples whose data is included in the repo don't even include this section, so you might consider removing the section altogether.

File: arcgis-ios-sdk-samples/Layers/Colormap renderer/README.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
+# Colormap Renderer
+
+Apply a colormap renderer to a raster.
+
+![Colormap renderer](colormap-renderer.png)
+
+## Use Case
+
+A colormap renderer transforms pixel values in a raster to display raster data based on specific colors, aiding in visual analysis of the data. For example, a forestry commission may want to quickly visualize areas above and below the tree-line occurring at a know elevation on a raster containing elevation values. They could overlay a transparent colormap set to color those areas below the tree-line elevation green, and those above white.
```
Comment (sbaskaran) [inline arcgis-ios-sdk-samples/Layers/Colormap renderer/README.md]: "at a know elevation" -> "at a known elevation"?

**Comment by vquach2404** [comment]
Ready for re-review

File: arcgis-ios-sdk-samples/Layers/Colormap renderer/README.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+## How to use the sample
+
+Pan and zoom to explore the effect of the colormap applied to the raster.
+
+## How it works
+
+1. Create a raster from a raster file.
+2. Create a raster layer from the raster.
+3. Create an array of 250 colors to represent the possible values from the raster. Colors at the beginning of the list replace the darkest values in the raster and colors at the end of the list replaced the brightest values of the raster.
+4. Create a ColormapRenderer with the color array and apply it to the raster layer.
```
Comment (philium) [inline arcgis-ios-sdk-samples/Layers/Colormap renderer/README.md]:
```suggestion
4. Create a color map renderer with the color array and apply it to the raster layer.
```

**Pull request merged**